src/models/insight.ts

```diff
@@ -6,13 +6,8 @@ export type Evidence<EvidenceSource extends IEvidenceSource> = {
 	reason: string;
 };
 
-// An EvidenceCollection's fields contain different types of Evidence
-export interface IEvidenceCollection {
-	[evidenceType: string]: Array<Evidence<IEvidenceSource>>;
-}
-
 // An insight is a value about a business, with evidence for how it was inferred
-export type Insight<Value, EvidenceCollection extends IEvidenceCollection> = {
+export type Insight<Value, EvidenceCollection extends Record<string, Array<Evidence<IEvidenceSource>>>> = {
```

Comment:
We can restrict the possible keys for the `EvidenceCollection`, defining valid types of evidence. See https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype.

Reply:
We do restrict it by defining a specific concrete `EvidenceCollection` type to pass in as a generic parameter. `Insight` shouldn't really have to care about what the restrictions are